File: repos_estudiantes_ALVAREZ_primera_entrega/CORY_CORNEJO_ALEJANDRO/Polimorfismo/Ejer15/Part2E15/src/main/java/com/mycompany/part2e15/Laboratorio.java
package com.mycompany.part2e15;
import java.util.ArrayList;

public class Laboratorio {
    private String nombre;
    private int capacidad;
    private ArrayList<Ordenador> ordenadores;

    public Laboratorio(String nombre, int capacidad) {
        this.nombre = nombre;
        this.capacidad = capacidad;
        this.ordenadores = new ArrayList<>();
    }

    public String getNombre() { return nombre; }
    public ArrayList<Ordenador> getOrdenadores() { return ordenadores; }

    public void agregarOrdenador(Ordenador o) {
        if (ordenadores.size() < capacidad) {
            ordenadores.add(o);
        } else {
            System.out.println("Capacidad máxima alcanzada en " + nombre);
        }
    }

    // Sobrecarga del método informacion
    public void informacion() {
        System.out.println("\nTodos los ordenadores en " + nombre + ":");
        for (Ordenador o : ordenadores) o.mostrar();
    }

    public void informacion(String estado) {
        System.out.println("\nOrdenadores en estado '" + estado + "' en " + nombre + ":");
        for (Ordenador o : ordenadores) {
            if (o.getEstado().equalsIgnoreCase(estado)) o.mostrar();
        }
    }

    public void informacion(int ramMinima) {
        System.out.println("\nOrdenadores con más de " + ramMinima + " GB de RAM en " + nombre + ":");
        for (Ordenador o : ordenadores) {
            if (o.getRam() > ramMinima) o.mostrar();
        }
    }

    public void mostrarSeriales() {
        System.out.print(nombre + " → Seriales: ");
        for (Ordenador o : ordenadores) {
            System.out.print(o.getSerial() + " ");
        }
        System.out.println();
    }

    public void eliminarPorSerial(String serial) {
        ordenadores.removeIf(o -> o.getSerial().equals(serial));
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/ARUQUIPA_VARGAS_ANA_GLORIA/POLIMORFISMO/Ejercicio15/Laboratorio.py

from Ordenador import Ordenador

class Laboratorio:
    def __init__(self, nombre, capacidad):
        self.nombre = nombre
        self.capacidad = capacidad
        self.ordenadores = []

    def agregar_ordenador(self, ordenador):
        if len(self.ordenadores) < self.capacidad:
            self.ordenadores.append(ordenador)
        else:
            print(f" Capacidad máxima alcanzada en {self.nombre}")

    def informacion(self, criterio=None):
        print(f"\n Información del laboratorio {self.nombre}")
        if criterio == "activo" or criterio == "inactivo":
            print(f" Ordenadores en estado '{criterio}':")
            for o in self.ordenadores:
                if o.estado == criterio:
                    o.mostrar()
        elif criterio == "ram>8":
            print(" Ordenadores con RAM > 8 GB:")
            for o in self.ordenadores:
                if o.ram > 8:
                    o.mostrar()
        else:
            print("Todos los ordenadores:")
            for o in self.ordenadores:
                o.mostrar()
